### PR TITLE
Support edit yaml for catalog without questions

### PR DESCRIPTION
**Pull request by loganhz**
https://github.com/rancher/rancher/issues/13064
https://github.com/rancher/rancher/issues/9771

image
image

### CODE REVIEW

**Comment by loganhz** [comment]
FYI @vincent99 conflits resolved

**Comment by vincent99** [comment]
This is kind of weird... it works with ones that do have questions too but doesn't convert bools correctly, it turns structures into dotted keys, the cancel button is really more of an edit as a form vs YAML and doesn't cancel anything, read from a file doesn't seem to do anything in either mode..

**Comment by loganhz** [comment]
Hi @vincent99 ,

1.  it works with ones that do have questions too

     `input-custom-answers` is newly added in this PR for app without questions. The feature of  supporting edit yaml  for app with questions has been merged two weeks ago. 

2.  but doesn't convert bools correctly.

    the schema type of `answers` is `map[string]`.  So boolean doesn't look good in YAML format. But server only takes string.

3.  it turns structures into dotted keys

     the schema type of `answers` is `map[string]`. So it only takes dotted keys. Server doesn't support answers in json format or YAML format now.

4. the cancel button is really more of an edit as a form vs YAML and doesn't cancel anything

    Updated. Renamed it to `Edit as YAML` and `Edit as a Form`.

5. read from a file doesn't seem to do anything in either mode.

    It should be working. I have tested it with the following yaml.

```
metrics:
  enabled: false
  image: wrouesnel/postgres_exporter
  imageTag: v0.1.1
  imagePullPolicy: IfNotPresent
  resources:
    requests:
      memory: 256Mi
      cpu: 100m
```

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/9772

**Comment by westlywright** [comment]
@vincent99 are we good to move forward with this?

**Comment by vincent99** [comment]
@westlywright I haven't looked at it since changes were made, but yes if it works consistently

**Comment by westlywright** [comment]
@loganhz I checked this out this morning to play with it and I saw a couple issues. If I use the `Edit as Yaml` on the `hl-couchdb` catalog app I get a few errors.

1. Using `tab` causes an uncaught error about tabs not being allowed. The driver-rke component uses the same yaml component as this and does not fire the same error when using the tab key. It looks like we handle the tab error on that page. 
2. While editing the file I get a lot of `unable to parse` uncaught exceptions. I'm not 100% sure why we dont get these on the rke page but again I'd prob investigate how that was implemented to understand why.
3. Rebase when you figure these out

File: app/components/input-custom-answers/component.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  intl:          service(),
+  layout,
+  pasteOrUpload: false,
+  showInput:     true,
+  accept:        '.yml, .yaml',
+  app:           null,
+  _boundChange:  null,
+
+  didInsertElement() {
+    this.set('_boundChange', (event) => {
```
Comment (westlywright) [inline app/components/input-custom-answers/component.js]: `this.set` => `set(this,`

File: app/components/input-custom-answers/component.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    this.$('INPUT[type=file]').on('change', this.get('_boundChange'));
+  },
+
+  actions: {
+    upload() {
+      this.$('INPUT[type=file]')[0].click();
+    },
+
+    showPaste() {
+      this.set('pasteOrUpload', true);
```
Comment (westlywright) [inline app/components/input-custom-answers/component.js]: this.set => set(this,

File: app/components/input-custom-answers/component.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    },
+
+    cancel() {
+      this.set('pasteOrUpload', false);
```
Comment (westlywright) [inline app/components/input-custom-answers/component.js]: this.set => set(this,

File: app/components/input-custom-answers/component.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+  },
+
+  actualAccept: computed('accept', function() {
+    if ( isSafari ) {
+      return '';
+    } else {
+      return this.get('accept');
```
Comment (westlywright) [inline app/components/input-custom-answers/component.js]: this.get => get(this,

**Comment by loganhz** [comment]
@westlywright Updated! Thanks for your review

**Pull request merged**